### PR TITLE
Add short link warnings and tidy up AIS-208

### PR DESCRIPTION
**Pull request by E-B-Smith**
* Use the cached userAgentString when possible.  
* Organize BranchUniversalObject.m code by functionality.  
* Add short link failure warning.

### CODE REVIEW

**Comment by BranchMacMini** [comment]
Result of [Integration 1](https://stlt.herokuapp.com/v1/xcs_deeplink/BranchMacMini.local/ea5ae53b09b7154843780a0f5b04267f/ea5ae53b09b7154843780a0f5b0457cf)
---
*Duration*: 8 minutes and 27 seconds
*Result*: **1 error, failing state: build-errors**

File: Branch-SDK/Branch-SDK/Requests/BranchShortUrlSyncRequest.m [hunk 1]
```diff
@@ -64,6 +64,8 @@ - (BNCServerResponse *)makeRequest:(BNCServerInterface *)serverInterface key:(NS
 
 - (NSString *)processResponse:(BNCServerResponse *)response {
     if (![response.statusCode isEqualToNumber:@200]) {
+        NSLog(@"Short link creation received HTTP status code %@.", response.statusCode);
```
Comment (ahmednawar) [inline Branch-SDK/Branch-SDK/Requests/BranchShortUrlSyncRequest.m]: Thoughts on using the existing `logWarning` method here?

File: Branch-SDK/Branch-SDK/BranchActivityItemProvider.m [hunk 2]
```diff
@@ -39,7 +40,7 @@ - (id)initWithParams:(NSDictionary *)params tags:(NSArray *)tags feature:(NSStri
         _stage = stage;
         _campaign = campaign;
         _alias = alias;
-        _userAgentString = [[[UIWebView alloc] init] stringByEvaluatingJavaScriptFromString:@"navigator.userAgent"];
+        _userAgentString = [BNCDeviceInfo userAgentString];
```
Comment (ahmednawar) [inline Branch-SDK/Branch-SDK/BranchActivityItemProvider.m]: <3

File: Branch-SDK/Branch-SDK/Requests/BranchShortUrlSyncRequest.m (hunk 1 above)
Comment (E-B-Smith) [inline Branch-SDK/Branch-SDK/Requests/BranchShortUrlSyncRequest.m]: I'm hoping to swap out all the `NSLog` and `logWarning` with the new `BNCLog` functions in the near future.

**Comment by ahmednawar** [comment]
👍

**Comment by BranchMacMini** [comment]
Result of [Integration 2](https://stlt.herokuapp.com/v1/xcs_deeplink/BranchMacMini.local/1d6f071064e67f7d4dee3ef2e1434cd7/1d6f071064e67f7d4dee3ef2e143555d)
---
*Duration*: 2 minutes and 19 seconds
*Result*: **Perfect build!** All 143 tests passed. :+1:
*Test Coverage*: 60%

**Pull request merged**